src/server/search/search_family.cc

```diff
@@ -241,43 +258,50 @@ void ReplyWithResults(const SearchParams& params, absl::Span<SearchResult> resul
   }
 }
 
-void ReplyKnn(size_t knn_limit, string_view knn_score_alias, const SearchParams& params,
-              absl::Span<SearchResult> results, ConnectionContext* cntx) {
+void ReplySorted(search::AggregationInfo agg, const SearchParams& params,
+                 absl::Span<SearchResult> results, ConnectionContext* cntx) {
+  size_t total = 0;
   vector<SerializedSearchDoc*> docs;
   for (auto& shard_results : results) {
+    total += shard_results.total_hits;
     for (auto& doc : shard_results.docs) {
       docs.push_back(&doc);
     }
   }
 
-  size_t prefix = min(params.limit_offset + params.limit_total, knn_limit);
+  size_t agg_limit = agg.limit.value_or(total);
+  size_t prefix = min(params.limit_offset + params.limit_total, agg_limit);
 
-  partial_sort(docs.begin(), docs.begin() + min(docs.size(), prefix), docs.end(),
-               [](const auto* l, const auto* r) { return l->knn_distance < r->knn_distance; });
-  docs.resize(min(docs.size(), knn_limit));
+  if (agg.descending)
+    partial_sort(docs.begin(), docs.begin() + min(docs.size(), prefix), docs.end(),
+                 [](const auto* l, const auto* r) { return *l > *r; });
+  else
+    partial_sort(docs.begin(), docs.begin() + min(docs.size(), prefix), docs.end(),
+                 [](const auto* l, const auto* r) { return *l < *r; });
```

Comment:
```suggestion
    partial_sort(docs.begin(), docs.begin() + min(docs.size(), prefix), docs.end(),
                 agg.descending
                     ? +[](const auto* l, const auto* r) { return *l < *r; }
                     : +[](const auto* l, const auto* r) { return *l > *r; }));
```

(just a suggestion, do as you will)

Reply:
I wish I could do this, but this is c++. Those lambdas are have different unique types - they can't be passed as one

Reply:
That's why I added the `+` sign - it converts the (captureless) lambda to a global function pointer, I think that should work. Doesn't it?

Reply:
True but I can also just capture the descending conditional, this is usually called for a few dozens of entries so performance is not critical